### PR TITLE
Fix: missing heading for intersections

### PR DESCRIPTION
**Pull request by Neyromancer**
Originally only 2 edges were added along the route after the fix any possible number of intersecting edges are added to the final route.

### CODE REVIEW

**Comment by Neyromancer** [comment]
Tets to be added

**Comment by Neyromancer** [comment]
Tests added

File: src/tyr/route_serializer_osrm.cc [hunk 1]
```diff
@@ -549,6 +549,14 @@ json::ArrayPtr intersections(const valhalla::DirectionsLeg::Maneuver& maneuver,
       edges.emplace_back(((prior_heading + 180) % 360), entry, true, false);
     }
 
+    if (i > 0 && !arrive_maneuver) {
+      bool entry = (arrive_maneuver) ? true : false;
```
Comment (merkispavel) [inline src/tyr/route_serializer_osrm.cc:553]: 1. `arrive_maneuver` is always equal to `false` in this branch so this variable/ternary operator is redundant
2. The entry is set to false if the intersecting edge is not routable in that direction. I'm not sure that is should be always false..

File: src/tyr/route_serializer_osrm.cc [hunk 2]
```diff
@@ -549,6 +549,14 @@ json::ArrayPtr intersections(const valhalla::DirectionsLeg::Maneuver& maneuver,
...
+      for (uint32_t m = 0; m < node->intersecting_edge_size(); m++) {
+        auto intersecting_edge = node->GetIntersectingEdge(m);
+        edges.emplace_back(intersecting_edge->begin_heading(), entry, true, false);
```
Comment (merkispavel) [inline src/tyr/route_serializer_osrm.cc:556]: as these are outgoing edges - boolean flags should be set vice versa, nope?
```suggestion
        edges.emplace_back(intersecting_edge->begin_heading(), entry, false, true);
```

File: src/tyr/route_serializer_osrm.cc [hunk 3]
```diff
@@ -549,6 +549,14 @@ json::ArrayPtr intersections(const valhalla::DirectionsLeg::Maneuver& maneuver,
       edges.emplace_back(((prior_heading + 180) % 360), entry, true, false);
     }
 
+    if (i > 0 && !arrive_maneuver) {
```
Comment (merkispavel) [inline src/tyr/route_serializer_osrm.cc:552]: I would at it to the code where "main" outgoing edge is added
```cpp
// Add the edge departing the node
    if (!arrive_maneuver) {
      edges.emplace_back(curr_edge->begin_heading(), true, false, true);
      // add other outgoing edges
      ...
    }
```

**Review by merkispavel — CHANGES_REQUESTED** [review]

File: src/tyr/route_serializer_osrm.cc (hunk 3 above)
Comment (Neyromancer) [reply src/tyr/route_serializer_osrm.cc:552]: the logic here is different, we are adding intersecting edges that are not starting or not ending, only in the middle of the road.

File: src/tyr/route_serializer_osrm.cc [hunk 4]
```diff
@@ -532,16 +532,23 @@ json::ArrayPtr intersections(const valhalla::DirectionsLeg::Maneuver& maneuver,
...
+    // the arrival step.
     std::vector<IntersectionEdges> edges;
 
     // Add the edge departing the node
     if (!arrive_maneuver) {
       edges.emplace_back(curr_edge->begin_heading(), true, false, true);
+      if (i > 0) {
+        for (uint32_t m = 0; m < node->intersecting_edge_size(); m++) {
+          auto intersecting_edge = node->GetIntersectingEdge(m);
+          bool routable = intersecting_edge->IsTraversableOutbound(curr_edge->travel_mode());
```
Comment (merkispavel) [inline src/tyr/route_serializer_osrm.cc:544]: wow, what a nice method. Exactly what we need!

**Review by merkispavel — APPROVED** [review]
Nice, looks good

**Review by merkispavel — APPROVED** [review]
Re approve after merging master with conflicts

**Pull request merged**